hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/debug/ContainerKeyScanner.java

```diff
@@ -0,0 +1,421 @@
+  private static final String KEY_TABLE = "keyTable";
+  private static final String DIRECTORY_TABLE = "directoryTable";
+
+  @CommandLine.Spec
+  private static CommandLine.Model.CommandSpec spec;
+
+  @CommandLine.ParentCommand
+  private RDBParser parent;
+
+  @CommandLine.Option(names = {"-ids", "--container-ids"},
+      split = ",",
+      paramLabel = "containerIDs",
+      required = true,
+      description = "Set of container IDs to be used for getting all " +
+          "their keys. Example-usage: 1,11,2 (Separated by ',').")
+  private Set<Long> containerIds;
+
+  private static Map<String, OmDirectoryInfo> directoryTable;
+  private static boolean isDirTableLoaded = false;
+
+  @Override
+  public Void call() throws Exception {
+    ContainerKeyInfoWrapper containerKeyInfoWrapper =
+        scanDBForContainerKeys(parent.getDbPath());
+
+    printOutput(containerKeyInfoWrapper);
+
+    closeStdChannels();
+
+    return null;
+  }
+
+  private void closeStdChannels() {
+    out().close();
+    err().close();
+  }
+
+  private Map<String, OmDirectoryInfo> getDirectoryTableData(String dbPath)
+      throws RocksDBException, IOException {
+    Map<String, OmDirectoryInfo> directoryTableData = new HashMap<>();
+
+    // Get all tables from RocksDB
+    List<ColumnFamilyDescriptor> columnFamilyDescriptors =
+        RocksDBUtils.getColumnFamilyDescriptors(dbPath);
+    final List<ColumnFamilyHandle> columnFamilyHandles = new ArrayList<>();
+
+    // Get all table handles
+    try (ManagedRocksDB db = ManagedRocksDB.openReadOnly(dbPath,
+        columnFamilyDescriptors, columnFamilyHandles)) {
+      dbPath = removeTrailingSlashIfNeeded(dbPath);
+      DBDefinition dbDefinition = DBDefinitionFactory.getDefinition(
+          Paths.get(dbPath), new OzoneConfiguration());
+      if (dbDefinition == null) {
+        throw new IllegalStateException("Incorrect DB Path");
+      }
+
+      // Get directory table
+      DBColumnFamilyDefinition<?, ?> columnFamilyDefinition =
+          dbDefinition.getColumnFamily(DIRECTORY_TABLE);
+      if (columnFamilyDefinition == null) {
+        throw new IllegalStateException(
+            "Table with name" + DIRECTORY_TABLE + " not found");
+      }
+
+      // Get directory table handle
+      ColumnFamilyHandle columnFamilyHandle = getColumnFamilyHandle(
+          columnFamilyDefinition.getName().getBytes(UTF_8),
+          columnFamilyHandles);
+      if (columnFamilyHandle == null) {
+        throw new IllegalStateException("columnFamilyHandle is null");
+      }
+
+      // Get iterator for directory table
+      try (ManagedRocksIterator iterator = new ManagedRocksIterator(
+          db.get().newIterator(columnFamilyHandle))) {
+        iterator.get().seekToFirst();
+        while (iterator.get().isValid()) {
+          directoryTableData.put(new String(iterator.get().key(), UTF_8),
+              ((OmDirectoryInfo) columnFamilyDefinition.getValueCodec()
```

Comment:
Similar to previous comment, it can be `OmDirectoryInfo.getCodec().fromPersistedFormat(iterator.get().value())`.

Reply:
Fixed.